PWGHF/HFC/Utils/utilsCorrelations.h

```diff
@@ -0,0 +1,80 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file utilsCorrelations.h
+/// \brief Xu Wang <waxu@cern.ch>
+
+#ifndef PWGHF_HFC_UTILS_UTILSCORRELATIONS_H_
+#define PWGHF_HFC_UTILS_UTILSCORRELATIONS_H_
+
+#include <cmath>
+
+namespace o2::analysis::hf_correlations
+{
+enum Region {
+  Default = 0,
+  Toward,
+  Away,
+  Transverse
+};
+
+template <typename T>
+Region getRegion(T deltaPhi)
+{
+  if (std::abs(deltaPhi) < o2::constants::math::PI / 3.) {
+    return Toward;
+  } else if (deltaPhi > 2. * o2::constants::math::PI / 3. && deltaPhi < 4. * o2::constants::math::PI / 3.) {
+    return Away;
+  } else {
+    return Transverse;
+  }
+}
+
+// Find Leading Particle
+template <typename TTracks>
+int findLeadingParticle(TTracks const& tracks)
+{
+  auto leadingParticle = tracks.begin();
+  for (auto const& track : tracks) {
+    if (std::abs(track.dcaXY()) >= 1. || std::abs(track.dcaZ()) >= 1.) {
+      continue;
+    }
+    if (track.pt() > leadingParticle.pt()) {
+      leadingParticle = track;
+    }
+  }
+  int leadingIndex = leadingParticle.globalIndex();
+  return leadingIndex;
+}
+
+// ======= Find Leading Particle for McGen ============
+template <typename TMcParticles, typename T1, typename T2>
+int findLeadingParticleMcGen(TMcParticles const& mcParticles, T1 etaTrackMax, T2 ptTrackMin)
```

Comment:
```suggestion
int findLeadingParticleMcGen(TMcParticles const& mcParticles, T1 const etaTrackMax, T2 const ptTrackMin)
```

Reply:
Sorry,When I use const, I encounter this kind of error.
/home/wangxu/alice/sw/SOURCES/O2Physics/master/0/PWGHF/HFC/Utils/utilsCorrelations.h:63:38: error: no match for 'operator>' (operand types are 'float' and 'const o2::framework::Configurable<float>')
   63 |       if (std::abs(mcParticle.eta()) > etaTrackMax)

Reply:
When I added .value(), it still produced an error.
/home/wangxu/alice/sw/SOURCES/O2Physics/master/0/PWGHF/HFC/Utils/utilsCorrelations.h:63:57: error: expression cannot be used as a function
   63 |       if (std::abs(mcParticle.eta()) > etaTrackMax.value())

Reply:
I have referenced other people's code, which uses float.
 https://github.com/AliceO2Group/O2Physics/blob/master/PWGHF/Utils/utilsAnalysis.h#L49

Reply:
You added `()` which is wrong.

Reply:
You are very impressive, and I am truly grateful. I have solved it.